velox/exec/HashTable.cpp

```diff
@@ -1709,8 +1712,20 @@ template <bool ignoreNullKeys>
 void HashTable<ignoreNullKeys>::prepareJoinTable(
     std::vector<std::unique_ptr<BaseHashTable>> tables,
     int8_t spillInputStartPartitionBit,
-    folly::Executor* executor) {
+    folly::Executor* executor,
+    bool dropDuplicates) {
   buildExecutor_ = executor;
+  if (dropDuplicates) {
```

Comment:
At this point the dedup could have possibly been abandoned?

Reply:
Yes, but we still need to clean up `table_`, because as long as HashBuild has ever entered the deduplication process, we need to perform rehash.